### ISSUE TITLE
Stream arguments passed by reference

### ISSUE THREAD

**Issue by rakhimov** [theandrewdavis/cpplint#5]
Non-const reference arguments for functions are violation of the guideline.
For output arguments, the pointers are requested,
but passing streams with pointers is awkward.

```cpp
void Print(std::ostream* out) { (*out) << "text"; }
Print(&std::cout);
```
vs.
```cpp
void Print(std::ostream& out) { out << "text"; }
Print(std::cout);
```
Cpplint only excuses streams by reference on overloading the ``operator<<``.
It would also be helpful to completely exclude streams from the rule of pass variable only by const reference.

**Comment by theandrewdavis** [comment]
Thanks for the issue, rakhimov! I definitely agree that having to dereference pointers to streams is awkward, but I think that's more of a subjective distaste. Since the style guide doesn't call out streams as an exception, I think it's better to keep the original enforcement policy.

Fortunately, it's pretty easy to ignore this warning. You can add `--filter=-runtime/references` to cpplint at the command line, or you can add a comment to ignore the lint warning like this:

    void Print(std::ostream* out) { (*out) << "text"; } // NOLINT(runtime/references)

or this:

    // NOLINTNEXTLINE(runtime/references)
    void Print(std::ostream* out) { (*out) << "text"; }

Hope that helps!
